portal/app/[locale]/_components/appLayout.tsx

```diff
@@ -1,82 +1,51 @@
 'use client'
 
-import { useNetworkType } from 'hooks/useNetworkType'
-import { useOnClickOutside } from 'hooks/useOnClickOutside'
-import { usePathnameWithoutLocale } from 'hooks/usePathnameWithoutLocale'
-import React, {
-  Dispatch,
-  MouseEventHandler,
-  SetStateAction,
-  Suspense,
-  useEffect,
-  useState,
-} from 'react'
+import { DrawerLoader } from 'components/drawer/drawerLoader'
+import dynamic from 'next/dynamic'
+import React, { Suspense, useState } from 'react'
 
 import { AppLayoutContainer } from './appLayoutContainer'
 import { Header } from './header'
 import { MainContainer } from './mainContainer'
-import { Navbar } from './navbar'
+import { NavBarUrlSync } from './navbar/navBarUrlSync'
 import { TestnetIndicator } from './testnetIndicator'
 
-type Props = {
-  children: React.ReactNode
-}
-
-const Backdrop = ({
-  onClick,
-}: {
-  onClick: MouseEventHandler<HTMLDivElement>
-}) => (
-  <div
-    className="absolute left-0 top-0 z-20
-    h-screen w-screen bg-gradient-to-b
-    from-neutral-950/0 to-neutral-950/25
-    lg:hidden"
-    onClick={onClick}
-  />
+const NavbarResponsive = dynamic(
+  () => import('./navbar/navbarResponsive').then(mod => mod.NavbarResponsive),
+  {
+    loading: () => (
+      <DrawerLoader
+        className="h-90dvh md:w-54 w-full md:h-full"
+        position="left"
+      />
+    ),
+    ssr: false,
+  },
 )
 
-// UI-less component so I can wrap it on suspense.
-// Hooks can't be wrapped...
-const NavBarUrlSync = function ({
-  setIsMenuOpen,
-}: {
-  setIsMenuOpen: Dispatch<SetStateAction<boolean>>
-}) {
-  const [networkType] = useNetworkType()
-  const pathname = usePathnameWithoutLocale()
-
-  useEffect(
-    function closeNavBarOnUrlChange() {
-      setIsMenuOpen(false)
-    },
-    [networkType, pathname, setIsMenuOpen],
-  )
-
-  return null
+type Props = {
+  children: React.ReactNode
```

Comment:
You could import React on top of the file

```suggestion
  children:ReactNode
```